packages/vite/src/plugins/paths.ts

```diff
@@ -0,0 +1,27 @@
+import MagicString from 'magic-string'
+import type { Plugin } from 'vite'
+
+export interface RuntimePathsOptions {
+  sourcemap?: boolean
+}
+
+export function runtimePathsPlugin (options: RuntimePathsOptions): Plugin {
+  return {
+    name: 'nuxt:runtime-paths-dep',
+    enforce: 'post',
+    transform (code, id) {
+      if (code.includes('__VITE_ASSET__') || code.includes('__VITE_PUBLIC_ASSET__')) {
```

Comment:
Small perf: this technically iterates over characters of code (how long it is), twice while a compiled regex can iterate once and being faster...